saleor/product/utils/product_variants.py

```diff
@@ -0,0 +1,26 @@
+from django.core.exceptions import ValidationError
+
+from ..error_codes import ProductErrorCode
+
+
+def product_variant_exist(product):
+    if product.variants.first() is None:
+        raise ValidationError(
+            {
+                "product": ValidationError(
+                    f"Cannot add product `{product.id}` without variant.",
```

Comment:
I wondering if error message shouldn't be more generic, like: `Cannot manage product '{product.id}' without variant.`